internal/server/stat/stat_util.go

```diff
@@ -248,3 +249,35 @@ func getMetadataHash(m *pb.StatMetadata) uint32 {
 	}, "-")))
 	return h.Sum32()
 }
+
+// getSourceSeriesKey computes the key for *pb.SourceSeries.
+// The key is based on the unique properties of the observations.
+func getSourceSeriesKey(s *pb.SourceSeries) string {
+	return fmt.Sprintf(
+		"%s_%s_%s_%s_%s",
+		s.ImportName,
+		s.MeasurementMethod,
+		s.ObservationPeriod,
+		s.ScalingFactor,
+		s.Unit)
+}
+
+// MergeSourceSeries merges lists of SourceSeries. For same source series, the one
+// with the latest data will be used.
+func MergeSourceSeries(seriesList ...[]*pb.SourceSeries) []*pb.SourceSeries {
+	result := []*pb.SourceSeries{}
+	resultMap := map[string]*pb.SourceSeries{}
+	for _, series := range seriesList {
+		for _, s := range series {
+			key := getSourceSeriesKey(s)
+			if _, ok := resultMap[key]; ok && len(s.Val) < len(resultMap[key].Val) {
```

Comment:
nit: The comment above says latest, but we are checking longest?   Clarify in the comment above that for a given series, longest series often has the latest data too...